### PR TITLE
GPT4All Java Bindings and supported models list

### PR DESCRIPTION
**Pull request by raulraja**
Fixes https://github.com/xebia-functional/xef/issues/215

This PR introduces the official GPT4ALL Java bindings, which provide access to GPT4All functionality. With the introduction of these bindings, we no longer need to maintain the `llama.cpp` binaries. The changes in this PR remove references to `llama.cpp` and replace them with the new GPT4ALL Java bindings.

The key changes in this PR are as follows:

- Added the `Gpt4AllModel` class, which represents a GPT4All model and provides methods to fetch supported models from a JSON file.
- Removed the `GPT4AllModel` interface, as it is no longer needed.
- Removed the `LLModel` interface and its implementations (`LlamaModel` and `GPTJModel`), as they are no longer used.
- Updated the `GPT4All` interface to use the new GPT4ALL Java bindings.
- Modified the `GPT4All.invoke()` function to load the GPT4All model using the new bindings.
- Updated the `createCompletion()` and `createChatCompletion()` functions in the `GPT4All` interface to use the new GPT4ALL Java bindings for generating completions.
- Updated the `main()` function in the `Chat.kt` example to use the new GPT4ALL Java bindings and the `Gpt4AllModel.supportedModels()` function to display [the supported models](https://github.com/nomic-ai/gpt4all/blob/main/gpt4all-chat/metadata/models.json).
- Removed references to `llama.cpp` and related binaries from the codebase.
- Updated the build configuration to include the Kotlin Serialization plugin.

### CODE REVIEW

**Comment by raulraja** [comment]
@xebia-functional/team-ai

File: gpt4all-kotlin/src/jvmMain/kotlin/com/xebia/functional/gpt4all/Models.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  val parameters: String,
+  val quant: String,
+  val type: String,
+  val description: String,
+  val disableGUI: String? = null,
+  val url: String? = null
+) {
+  companion object {
+    private val url = "https://raw.githubusercontent.com/nomic-ai/gpt4all/main/gpt4all-chat/metadata/models.json"
+    fun supportedModels(): List<Gpt4AllModel> {
```
Comment (diesalbla) [inline gpt4all-kotlin/src/jvmMain/kotlin/com/xebia/functional/gpt4all/Models.kt:25]: Is this going to run the HTTP operation and fetch the list of supported models from that `URL` on every call?

Comment (raulraja) [reply gpt4all-kotlin/src/jvmMain/kotlin/com/xebia/functional/gpt4all/Models.kt:25]: For now, yes, We have discussed the concept of a Model Zoo / List that we support beyond the GPT4ALL ones with our curated list of models. Feel free to commit any improvements directly to this branch if we want to cache the results. I may not have much more time today to dedicate to it.

**Review by diesalbla — APPROVED** [review]

File: gpt4all-kotlin/src/jvmMain/kotlin/com/xebia/functional/gpt4all/Models.kt (hunk 1 above)
Comment (raulraja) [reply gpt4all-kotlin/src/jvmMain/kotlin/com/xebia/functional/gpt4all/Models.kt:25]: Let's address this when we have a strategy for our recommended models.

**Pull request merged**